packages/util-dynamodb/src/convertToAttr.ts

```diff
@@ -105,18 +109,19 @@ const convertToMapAttr = (
   data: { [key: string]: NativeAttributeValue },
   options?: marshallOptions
 ): { M: { [key: string]: AttributeValue } } => ({
-  M: Object.entries(data)
-    .filter(
-      ([key, value]: [string, NativeAttributeValue]) =>
-        !options?.removeUndefinedValues || (options?.removeUndefinedValues && value !== undefined)
-    )
-    .reduce(
-      (acc: { [key: string]: AttributeValue }, [key, value]: [string, NativeAttributeValue]) => ({
-        ...acc,
-        [key]: convertToAttr(value, options),
-      }),
-      {}
-    ),
+  M: (function getMapFromEnurablePropsInPrototypeChain(data) {
+    const map: { [key: string]: AttributeValue } = {};
+    for (const key in data) {
+      const value = data[key];
+      if (
+        typeof value !== "function" &&
+        (!options?.removeUndefinedValues || (options?.removeUndefinedValues && value !== undefined))
```

Comment:
Should this line be `(value !== undefined || !options?.removeUndefinedValues)`?

Reply:
The code right now is
```js
(!A || (A && B))
```

where:
* A: `options?.removeUndefinedValues`
* B: `value !== undefined`

It can be simplified to `(B || !A)`
Added in [`db48226` (#1974)](https://github.com/aws/aws-sdk-js-v3/pull/1974/commits/db482267db9fc1d65c08e5974f74d651d89eb476)